### PR TITLE
feat: add dashboard page

### LINKED ISSUE #26: [feature] add dashboard page
Add a dashboard page like: https://aws.amazon.com/blogs/security/deploy-dashboard-for-aws-waf-minimal-effort/ . It uses the logs as data input.

image

### PR DESCRIPTION
**Pull request by love98ooo**
Fix: https://github.com/casbin/caswaf/issues/26

### CODE REVIEW

File: controllers/metric.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"errors"
+	"github.com/casbin/caswaf/object"
```
Comment (nomeguy) [inline controllers/metric.go]: sort

File: controllers/metric.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	}
+	var total int64
+	for _, metric := range *metrics {
+		total += metric.Count
+	}
+	c.ResponseOk(metrics, total)
+}
+
+func granularity2TimeType(rangeType string) string {
+	switch rangeType {
```
Comment (nomeguy) [inline controllers/metric.go:47]: Is this same as Casibase?

File: controllers/metric.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	dtoType := c.Input().Get("type")
+	dataType, err := type2DataType(dtoType)
+	if err != nil {
+		c.ResponseError(err.Error())
+		return
+	}
+	rangeType := c.Input().Get("rangeType")
+	count, err := strconv.Atoi(c.Input().Get("count"))
```
Comment (nomeguy) [inline controllers/metric.go]: util.ParseInt for everywhere

File: web/src/DashboardPage.js [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+      },
+    };
+    return (
+      <Card title={i18next.t(`general:${title}`)}>
+        <ReactECharts option={option}></ReactECharts>
+      </Card>
+    );
+  }
+
+  renderBarChart(title, data) {
```
Comment (nomeguy) [inline web/src/DashboardPage.js:236]: Define a JS file for each card, put them into a folder

File: web/src/index.css [hunk 5]
```diff
@@ -30,6 +30,10 @@ code {
   padding: 2px !important;
 }
 
+.ant-card {
```
Comment (nomeguy) [inline web/src/index.css]: Can we avoid global CSS?

File: web/yarn.lock [hunk 6]
```diff
@@ -4110,6 +4110,22 @@ duplexer@^0.1.2:
   resolved "https://registry.yarnpkg.com/duplexer/-/duplexer-0.1.2.tgz#3abe43aef3835f8ae077d136ddce0f276b0400e6"
   integrity sha512-jtD6YG370ZCIi/9GTaJKQxWTZD045+4R4hTk/x1UyoqadyJ9x9CgSi1RlVDQF8U2sxLLSnFkCaMihqljHIWgMg==
 
+echarts-for-react@^3.0.2:
```
Comment (nomeguy) [inline web/yarn.lock:4113]: Use https://registry.yarnpkg.com for everywhere

**Review by nomeguy — CHANGES_REQUESTED** [review]

File: controllers/metric.go (hunk 2 above)
Comment (love98ooo) [reply controllers/metric.go:47]: They have similar functions, but different abstractions.

**Review by nomeguy — APPROVED** [review]

**Pull request merged**